### DISCUSSION TITLE
Best Practice: Global Site Data from a CMS

### DISCUSSION THREAD

**Discussion by ivoilic** [vercel/next.js#15306] [Help]
What is the best way to pull global site data (Site name, global nav data, etc) from a CMS?
Doesn't seem very efficient to pull that sort of data on every page.
The alternative is running a script to pull that data before the site builds and save it to a json file.
It feels like there should be a cleaner solution to this built into Next.

Does anyone have a better solution?

**Comment by jamesmosier** [comment]
Shared global data can be placed in [pages/_app.js](https://nextjs.org/docs/advanced-features/custom-app) using `getInitialProps`, although there are some downsides to using this as [mentioned here](https://nextjs.org/docs/api-reference/data-fetching/getInitialProps).

There may be some good info in some of the other discussions that touched on this subject: https://github.com/vercel/next.js/discussions?discussions_q=global+data
